src/ILLink.Shared/TrimAnalysis/RequireDynamicallyAccessedMembersAction.cs

```diff
@@ -22,7 +22,7 @@ public void Invoke (in DiagnosticContext diagnosticContext, in MultiValue value,
 					var availableMemberTypes = valueWithDynamicallyAccessedMembers.DynamicallyAccessedMemberTypes;
 					if (!Annotations.SourceHasRequiredAnnotations (availableMemberTypes, targetValue.DynamicallyAccessedMemberTypes, out var missingMemberTypes)) {
 						(var diagnosticId, var diagnosticArguments) = Annotations.GetDiagnosticForAnnotationMismatch (valueWithDynamicallyAccessedMembers, targetValue, missingMemberTypes);
-						diagnosticContext.AddDiagnostic (diagnosticId, diagnosticArguments);
+						_diagnosticContext.AddDiagnostic (diagnosticId, diagnosticArguments);
```

Comment:
Declare the `_diagnosticContext` field on the shared struct (so in this file) and as readonly. That will "force" the analyzer/linker to initialize it in the .ctors. Cleaner than relying on the fields existence from the "outside".